feature/troubleshooting/src/main/java/com/simprints/feature/troubleshooting/overview/usecase/ExportLogsUseCase.kt

```diff
@@ -0,0 +1,85 @@
+package com.simprints.feature.troubleshooting.overview.usecase
+
+import android.content.Context
+import com.simprints.core.DeviceID
+import com.simprints.feature.troubleshooting.FileNameDateTimeFormatter
+import com.simprints.infra.config.sync.ConfigManager
+import com.simprints.infra.logging.LogDirectoryProvider
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import net.lingala.zip4j.ZipFile
+import net.lingala.zip4j.model.ZipParameters
+import net.lingala.zip4j.model.enums.EncryptionMethod
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.Date
+import javax.inject.Inject
+
+internal class ExportLogsUseCase @Inject constructor(
+    @DeviceID private val deviceId: String,
+    @ApplicationContext private val context: Context,
+    private val logDirectoryProvider: LogDirectoryProvider,
+    private val configManager: ConfigManager,
+    @FileNameDateTimeFormatter private val dateFormatter: SimpleDateFormat,
+) {
+    operator fun invoke(): Flow<LogsExportResult> = flow {
+        emit(LogsExportResult.InProgress)
+        val directory = logDirectoryProvider(context)
+        val files = directory.listFiles().orEmpty()
+
+        if (files.isNotEmpty()) {
+            val archiveFile = createArchiveFile(dateFormatter.format(Date()) + "_$deviceId")
+            val password = getPassword()
+            createZipArchiveWithPassword(archiveFile, files.toList(), password)
+
+            delay(2000L) // To look like it does something
+            emit(LogsExportResult.Success(deviceId, archiveFile))
+        } else {
+            emit(LogsExportResult.NotStarted)
```

Comment:
Wouldn't this make users to tap the button again and again thinking it's not working?

Reply:
They might, but this is better than leaving it in a broken state.

Reply:
I will add a "failed" state to briefly show different text.

Reply:
Yeah, that's what I meant - maybe just show a toast "Nothing to export".